components/org.wso2.bfsi.consent.management.service/src/main/java/org/wso2/bfsi/consent/management/service/internal/ConsentManagementServiceComponent.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ * <p>
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.consent.management.service.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.bfsi.consent.management.common.config.ConsentManagementConfigParser;
+import org.wso2.bfsi.consent.management.common.exceptions.ConsentManagementRuntimeException;
+import org.wso2.bfsi.consent.management.common.persistence.JDBCPersistenceManager;
+import org.wso2.bfsi.consent.management.service.ConsentCoreService;
+import org.wso2.bfsi.consent.management.service.impl.ConsentCoreServiceImpl;
+import org.wso2.carbon.identity.oauth2.OAuth2Service;
+
+import java.sql.SQLException;
+
+/**
+ * Consent Management Core Service Component.
+ */
+@Component(name = "org.wso2.bfsi.consent.management.service.internal.ConsentManagementServiceComponent",
+        immediate = true)
+public class ConsentManagementServiceComponent {
+
+    private static final Log log = LogFactory.getLog(ConsentManagementServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        ConsentCoreService consentCoreService = new ConsentCoreServiceImpl();
+
+        // Verify BFSI consent database connection when the server starts up
+        try {
+            boolean isConnectionActive = JDBCPersistenceManager.getInstance().getDBConnection()
+                    .isValid(ConsentManagementConfigParser.getInstance().getConnectionVerificationTimeout());
+            if (!isConnectionActive) {
+                log.error("The connection is not active");
+                throw new ConsentManagementRuntimeException("The connection is not active");
+            }
+        } catch (SQLException e) {
+            log.error("Database connection is not active, cannot proceed");
+            throw new ConsentManagementRuntimeException("Database connection is not active, cannot proceed");
+        }
+
+
```

Comment:
```suggestion

```

Reply:
Fixed in Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/3/commits/08d281c66fc90f979dcae36a8c8baedf99ab9aa9